src/visibility.js

```diff
@@ -0,0 +1,173 @@
+    this.observer.observe(this.appTargetContainer);
+  }
+
+  _initFloating() {
+    const defaultFloatingConfig = {
+      floating: {
+        position: 'bottom-right',
+        size: {
+          height: '225px',
+          width: '400px'
+        },
+        margin: {
+          x: '2px',
+          y: '2px'
+        }
+      }
+    };
+    Utils.Object.mergeDeep(this.config, defaultFloatingConfig, Utils.Object.copyDeep(this.config));
+    this.floatingPoster = Utils.Dom.createElement('div');
+    this.floatingPoster.className = FLOATING_POSTER_CLASS;
+    this.floatingContainer = Utils.Dom.createElement('div');
+    this.floatingContainer.className = FLOATING_CONTAINER_CLASS;
+
+    this.appTargetContainer = Utils.Dom.getElementById(this.player.config.targetId);
+    this.floatingContainer.innerHTML = this.appTargetContainer.innerHTML;
+    this.appTargetContainer.innerHTML = '';
+    Utils.Dom.appendChild(this.appTargetContainer, this.floatingPoster);
+    Utils.Dom.appendChild(this.appTargetContainer, this.floatingContainer);
+    this.config.floating.position.split('-').forEach(side => {
+      Utils.Dom.addClassName(this.floatingContainer, `${FLOATING_ACTIVE_CLASS}-${side}`);
+    });
+  }
+
+  _dismissed() {
+    this.dismissed = true;
+    this.player.pause();
+    this._stopFloating();
+  }
+
+  _stopFloating() {
+    Utils.Dom.removeClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.removeAttribute(this.floatingContainer, 'style');
+  }
+
+  _startFloating() {
+    Utils.Dom.addClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.setStyle(this.floatingContainer, 'height', this.config.floating.size.height);
+    Utils.Dom.setStyle(this.floatingContainer, 'width', this.config.floating.size.width);
+    Utils.Dom.setStyle(this.floatingContainer, 'margin', `${this.config.floating.margin.y} ${this.config.floating.margin.x}`);
+  }
+
+  _handleVisibilityChange(entries: Array<window.IntersectionObserverEntry>) {
+    const playerIsOutOfVisibility = entries[0].intersectionRatio < this.config.threshold;
+    if (this.config.floating && this.everStartedPlaying && !this.dismissed) {
+      this._handleFloatingChange(playerIsOutOfVisibility);
+    }
+  }
+
+  _handleFloatingChange(playerIsOutOfVisibility: boolean) {
+    if (playerIsOutOfVisibility) {
+      this._startFloating();
+    } else {
+      this._stopFloating();
+    }
+  }
+  /**
+   * _addBindings
+   * @private
+   * @returns {void}
+   */
+  _addBindings(): void {
+    this.eventManager.listen(this.player, this.player.Event.FIRST_PLAYING, () => {
+      this.everStartedPlaying = true;
+      Utils.Dom.setStyle(this.floatingPoster, 'background-image', `url("${this.player.config.sources.poster}")`);
+    });
+  }
+
+  /**
+   * Destroys the plugin.
+   * @override
+   * @public
+   * @returns {void}
+   */
+  destroy(): void {
+    this.observer.disconnect();
```

Comment:
destroy event manager

Reply:
clear the player reference on destroying and make sure the local vars which suppose to be cleared on reset will be cleared

Reply:
@Yuvalke , why is it needed to clear the player reference on destroying? All the plugins have a reference to the player and don't clear it in the destroy. If it is needed it should be added in the basePlugin, but why?

Reply:
Correct should be added to `BasePlugin`.
Clear in destroy is needed for the garbage collector to be aware that this ref is not needed anymore.

Reply:
Basically it's needed to clear your local scope when you finish

Reply:
@RoyBregman please open a ticket for https://github.com/kaltura/playkit-js-visibility/pull/1#discussion_r439803539